### DISCUSSION TITLE
Is there a way to run our app on a mobile device or emulator directly without opening the platform's editor first?

### DISCUSSION THREAD

**Discussion by rhyek** [ionic-team/capacitor#4457] [Ideas]
My dev setup is a bit unorthodox. I have my editor running on a Macbook, but through ssh I am actually editing my project's source files on a remote linux minipc (on the same LAN) where I have configured android studio appropriately to work on flutter projects. I connect `adb` to my physical android phone via wifi (is listed when running `adb devices`) and so on that remote host I can just run `flutter run` (https://flutter.dev/docs/get-started/test-drive?tab=terminal) and it will launch the app on my phone wirelessly.

This is all being done for performance reasons (Macbooks generally have limited hardware specs). This still requires Android Studio to be configured on the remote host, but it works seamlessly and I never have to interact with it.

Currently I can't have the same workflow when using Capacitor since I have to do `npx cap open android` to open the editor first and then run the project from there. There should be a `npx cap run android` that works in a similar manner to `flutter run`.

**Comment by jcesarmobile** [comment]
Capacitor 3 has a `npx cap run` command, give it a try.

But make sure you read the [upgrade guide](https://capacitorjs.com/docs/v3/updating/3-0) first
